### ISSUE TITLE
Mocking input event for controlled inputs in react

### ISSUE THREAD

**Issue by briantingtc** [carlinyuen/ChromeAutoTextExpander#51]
This extension works in a way that leads to unexpected behaviour when used in a react controlled input. Currently this extension assigns the value to the dom element without dispatching/triggering an event: 
`app/scripts/ate.js`(ln.343)
``` javascript
textInput.value = replaceText(
  textInput.value,
  shortcut,
  autotext,
  cursorPosition
);
```
Here's a fiddle illustrating the problem: [JS Fiddle example](https://jsfiddle.net/r3b7tq6g/4/)

and a rudimentary fix I made to get this working:
https://github.com/briantingtc/ChromeAutoTextExpander/commit/76d8ad5b8b0bc17f4f195ead5bd2f42a5b15f1a4

Let me know if we could get something like the above commit merged in or whether or if adding custom behaviour for a specific domain would be possible.

**Comment by carlinyuen** [comment]
Hey Brian! Wow thanks, I've been meaning to get to this but since getting
married haven't had a chance to at all. I'll try to take a look tomorrow
and would love to merge it in if we can.

**Comment by briantingtc** [comment]
Thanks Carlin, let me know if there's anything I can do to help -- and congrats on getting hitched!

**Comment by wdoug** [comment]
Hey @carlinyuen we are running into this same issue with some of our users. Is there any way that we could help getting a fix merged?

**Comment by carlinyuen** [comment]
Hey Will, thanks for the poke. Yeah absolutely, I'm going to commit time to
looking at it this week, hopefully have the fix in by EOW at the latest.
Sorry for the wait!

.C

**Comment by carlinyuen** [comment]
Hey folks, confirming that I'm looking into this now. Thanks for being patient! Life got busy, not a great excuse. :P

**Comment by ghbeckman** [comment]
Hey @carlinyuen, wondering if you guys can push the most recent (2.0) version to the chrome web store.. I kinda rely on that tool for my daily activities and now with improvements in iframe support it would really help me out. thanks!

**Comment by carlinyuen** [comment]
Hey @ghbeckman workin' on it. :)
With the iframe change, I have to retest some of the code to make sure the extension still works in the same input fields. Hopefully can have this out for some of you folks by July 4th.

Actually, it'd really help to have a bunch of you test the changes for me on some of the sites you use. Please let me know if any of you would be interested in joining some sort of "dogfood" group. I'd create a Google group and basically send an early version of the extension to you, would just need you to test it on whatever product you already use it on, to make sure it's not busted.

**Comment by briantingtc** [comment]
I'd be down, lmk how I can help

**Comment by ghbeckman** [comment]
@carlinyuen Absolutely! Glad to help in anyway possible. Thanks for your effort. Appreciate it

**Comment by carlinyuen** [comment]
Hey folks, thanks for being great sports. Sincere apologies, life got in the way again. :( I'm going to try and get back into this this weekend.

**Comment by briantingtc** [comment]
Sorry to poke you again @carlinyuen as I know you're super busy but we'll be needing this plugin to work with one of our features. If I create a pr with specific behavior for our domain would you consider merging that in? Otherwise would it be okay if we re hosted the plugin with that specific change?

**Comment by dcfranca** [comment]
Any update on this? it's important for me thais fix as well

**Comment by ethelwulf13** [comment]
There is any plan to update to Manifest 3.0 ? This extension is great but now google says it may soon no longer be supported :(
